### DISCUSSION TITLE
Fetching JSON in Svelte's load() still shows up in Network Tab

### DISCUSSION THREAD

**Discussion by keehun** [vercel/community#713] [Help] [answered]
Hi Vercel Community,

I have a SvelteKit application. In one of my components in the `load()` function, I am reaching out to retrieve a JSON file. When the page first loads, I don't see the JSON file being requested in the browser since I assume SSR is correctly rendering this before serving it to the browser.

However, when I navigate to another page and then navigate back (all within the same application), I see that JSON file being requested by the browser and it shows up in the Network tab when I inspect the browser's traffic.

I assumed that `load()` would be called by SSR. I would like to obscure the JSON request if at all possible. Is this something I can enforce? The reason is the API contains data I would like to keep private to only those who should have access. By looking at the raw JSON, it would be possible for my visitors to crawl the API and scrape the entire dataset.

I should mention that I am using `adapter-auto` and am not doing any prerendering. The only "unusual" thing about my svelte.config.js is `trailingSlash: 'always'`

Thank you,
Keehun

**Comment by amyegan** [comment, marked as answer]
Hi @keehun. Thanks for reaching out! This sounds like this is more related to your Svelte setup than Vercel specifically. I'm going to try to help you with a few suggestions here, but you'll be able to get more detailed help from the Svelte experts on [SvelteKit Discussions](https://github.com/sveltejs/kit/discussions?discussions_q=load) and [Svelte Discord](https://svelte.dev/chat).

My understanding is that this is the default behavior for Svelte. Pages are server-side rendered, then re-rendered client-side for interactivity. There are some settings to help you control this behavior, which you can read more about it in [SvelteKit Page Options documentation](https://kit.svelte.dev/docs/page-options).

If you're trying to hide query parameters or the full data returned from your API, you could put the external API call in a serverless function [endpoint](https://kit.svelte.dev/docs/routing#endpoints) and [load](https://kit.svelte.dev/docs/loading) from that to obscure the actual API call and control what data gets returned to the page for rendering. This answer on Stack Overflow provides more explanation:  https://stackoverflow.com/a/67579049
